crypto/src/openssl.rs

```diff
@@ -177,6 +194,40 @@ impl Crypto for OpensslCrypto {
         Ok(cdi)
     }
 
+    fn get_exported_cdi_handle(
+        &mut self,
+        cdi: &Self::Cdi,
+    ) -> Result<ExportedCdiHandle, CryptoError> {
+        for (stored_cdi, stored_handle) in self.export_cdi_slots.iter() {
+            if stored_cdi == cdi {
+                return Ok(stored_handle.to_owned());
+            }
+        }
```

Comment:
Is this a valid scenario?

Reply:
Hm. It seems like a bug if the same CDI slot is used twice. We could return an error if it's already defined.

The only case where that wouldn't work is if the DPE code needs to retrieve the stored handle twice for some reason. But I don't think the current code does.

Reply:
Added code to make this an error, we can always backtrack this in case we need it later.